clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4769,31 +4769,32 @@ ExprResult Sema::CheckOSLogFormatStringArg(Expr *Arg) {
 /// Check that the user is calling the appropriate va_start builtin for the
 /// target and calling convention.
 static bool checkVAStartABI(Sema &S, unsigned BuiltinID, Expr *Fn) {
-  const llvm::Triple &TT = S.Context.getTargetInfo().getTriple();
+  const TargetInfo &TI = S.Context.getTargetInfo();
+  bool IsMicrosoftABI = TI.getCXXABI().isMicrosoft();
```

Comment:
This seems like it would potentially be an incorrect behavior change for mingw. cc @mstorsjo

Reply:
Indeed - mingw is Windows but does not use the Microsoft C++ ABI. It does use the Microsoft C ABI though, mostly.

Reply:
Thank you. I've updated the check to use TI.shouldUseMicrosoftCCforMangling which is true for all targets that use WindowsTargetInfo including MinGW. Please let me know where I should add a test that makes sure the mingw use case is correctly covered.